src/test/test_dynamoTable.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+from unittest.mock import MagicMock
+from ..main.dynamoTable import *
+from ..main.building import *
+from ..main.machine import *
+
+class TestDynamoTable:
+    # User table tests =========================================================
+    @pytest.yield_fixture(autouse=True)
+    def user_table(self):
+        userTable = DynamoTable('Users')

```

Comment:
It seems like you don't quite understand the concept of mocking. Which is completely fine, it took me a while to wrap my head around it as well. Right now, what this code is doing is creating a normal DynamoTable object, then in the test it's saying "Replace this method in the DynamoTable class with one that does nothing, then call it and make sure it was called".

Mocking is essentially a way of removing dependencies that are tested elsewhere, in this case the Dynamo resource from the AWS SDK.

The way we do this is by "injecting" the dependencies (you've probably heard of "dependency injection" before) in function parameters or class constructors. So in this case, we might make the DynamoTable `__init__` declaration like this:

```
def __init__(self, tableType, dynamodb=boto3.resource('dynamodb')):
    self.table = dynamodb.Table(tableType)
    ...
```

This doesn't change how it's used during production use, but it allows us to insert our own value for the aws dynamodb client. 

In this case, we could make a mock of that client and pass it in, and essentially say "when the Table method is called on this object, return a mock Table object". Then on that mock table, for each test we say "when this method is called on this mock table, return that. Then, verify that the output from the DynamoTable method matches what we expect, and that the method we expected to be called was called with the correct input."

For creating mocks of classes, check out https://docs.python.org/3/library/unittest.mock.html#unittest.mock.create_autospec and https://docs.python.org/3/library/unittest.mock.html#auto-speccing.

Also, lower down, where you create a mock user, that's not really necessary. You can just use a regular string, because we're not attempting to remove any dependency added by the string class.

Let me know if you have any questions.